controllers/gateway/reconcile_result.go

```diff
@@ -40,7 +40,6 @@ func (r *APIRuleReconciler) updateStatus(ctx context.Context, api *gatewayv1beta
 	api.Status.ObservedGeneration = api.Generation
 	api.Status.LastProcessedTime = &v1.Time{Time: time.Now()}
 
-	// TODO: api status as dependency injection
```

Comment:
This todo is still relevant for the status follow up

Reply:
Can we add more information to the follow up task instead of putting inline comments? Also the comment doesn't give much information. I sharing the idea in the follow up task what should be changed in what way will be more helpful.

Reply:
OK, agree :)